src/builtins/hmr-runtime.js

```diff
@@ -15,38 +15,43 @@ function Module() {
 module.bundle.Module = Module;
 
 if (!module.bundle.parent && typeof WebSocket !== 'undefined') {
-  var hostname = process.env.HMR_HOSTNAME || location.hostname;
-  var ws = new WebSocket('ws://' + hostname + ':' + process.env.HMR_PORT + '/');
-  ws.onmessage = function(event) {
-    var data = JSON.parse(event.data);
-
-    if (data.type === 'update') {
-      data.assets.forEach(function (asset) {
-        hmrApply(global.require, asset);
-      });
-
-      data.assets.forEach(function (asset) {
-        if (!asset.isNew) {
-          hmrAccept(global.require, asset.id);
-        }
-      });
-    }
+  try {
+    var hostname = process.env.HMR_HOSTNAME || location.hostname;
+    var ws = new WebSocket('ws://' + hostname + ':' + process.env.HMR_PORT + '/');
+    ws.onmessage = function(event) {
+      var data = JSON.parse(event.data);
+
+      if (data.type === 'update') {
+        data.assets.forEach(function (asset) {
+          hmrApply(global.require, asset);
+        });
+
+        data.assets.forEach(function (asset) {
+          if (!asset.isNew) {
+            hmrAccept(global.require, asset.id);
+          }
+        });
+      }
 
-    if (data.type === 'reload') {
-      ws.close();
-      ws.onclose = function () {
-        location.reload();
+      if (data.type === 'reload') {
+        ws.close();
+        ws.onclose = function () {
+          location.reload();
+        }
       }
-    }
 
-    if (data.type === 'error-resolved') {
-      console.log('[parcel] ✨ Error resolved');
-    }
+      if (data.type === 'error-resolved') {
+        console.log('[parcel] ✨ Error resolved');
+      }
 
-    if (data.type === 'error') {
-      console.error('[parcel] 🚨  ' + data.error.message + '\n' + 'data.error.stack');
-    }
-  };
+      if (data.type === 'error') {
+        console.error('[parcel] 🚨  ' + data.error.message + '\n' + data.error.stack);
+      }
+    };
+  } catch (e) {
```

Comment:
Do we need the try/catch around the entirety of this code or is only a certain line failing? Also what is the issue you had? Connection failure or something? What about a `ws.onerror` handler?

Reply:
I was using ngrok through https, so Chrome did not want to allow connection to non-secure websocket used by hmr. and it crashed my whole app.
for me, hmr can't crash the underlying app, that's why I made a (too?) broad try / catch.

Reply:
Isn't this fixed by the recent https enabled websocket?